### ISSUE TITLE
[python] SparseNDArray.write allocates excess memory when given Arrow ChunkedArrays

### ISSUE THREAD

**Issue by bkmartinjr** [single-cell-data/TileDB-SOMA#1896]
The current implementation path of SparseNDArray.write, for at least Arrow Table data, makes a full copy of the data when the values to write are Arrow ChunkedArray:

Code:
```
            data = values.column("soma_data").to_numpy()
            coord_tbl = values.drop(["soma_data"])
            coords = tuple(
                coord_tbl.column(f"soma_dim_{n}").to_numpy()
                for n in range(coord_tbl.num_columns)
            )
            arr[coords] = data
```

`to_numpy()` is not zero-copy for ChunkedArray (NumPy requires contiguous memory).  It is zero-copy for Arrow Array

ideally we would do all this with zero copy, having the underlying C++ layer handle multi-chunk array without an extra copy & flatten operation.

**Comment by johnkerl** [comment]
I'll need to reassign this to @nguyenv (or myself) for a subsequent sprint -- the write path is cppifying and then we can do copy-reduction optimizations as follow-ons.

**Comment by johnkerl** [comment]
@nguyenv is this completed?

**Comment by nguyenv** [comment]
No, we still call `to_numpy` here. This should be assigned to my current sprint - I'll move this down into the C++ later.

**Comment by bkmartinjr** [comment]
@nguyenv - the other pattern I've previously used is to convert the ChunkedArray to a tuple of array, and hand that to the C++ layer (e.g., `clib.some_function(tuple(chunk.to_numpy() for chunk in chunked_array.chunks))`).   Pybind handles both tuples and arrays with zero fuss, so this might be an option if it is easier than passing the Arrow object directly.
